### ISSUE TITLE
Add Login via Gmail Feature

### ISSUE THREAD

**Issue by SAITARUN55** [SAITARUN55/metroapp#2]

**Comment by sHiVaNgI821** [comment]
@SAITARUN55 I would like to work on this issue. Could you please elaborate on the expectations from this issue.

**Comment by shivaniaakanksha** [comment]
@SAITARUN55  I'm curious to work on this issue. May you please describe more regarding the issue.

**Comment by SAITARUN55** [comment]
@sHiVaNgI821 @shivaniaakanksha You both can work on the same issue but whichever is the best I will consider that finally. 
Basic Login Page with Email and Password with Social Integrations such as Gmail using Firebase.
In the Login Page, you have to create another activity for SignUp. Basic Details are enough from the user for Sign Up Page.

**Comment by shivaniaakanksha** [comment]
@SAITARUN55  After adding Firebase dependencies and setup for android  I'am facing this Error , written below.
Launching lib\main.dart on vivo 1601 in debug mode...
[!] Your app isn't using AndroidX.
    To avoid potential build failures, you can quickly migrate your app by following the steps on https://goo.gl/CP92wY.

FAILURE: Build failed with an exception.

* Where:
Build file 'E:\flutter\development\metroapp\android\app\build.gradle' line: 24

* What went wrong:
A problem occurred evaluating project ':app'.
> ASCII

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)

**Comment by SAITARUN55** [comment]
@shivaniaakanksha We have to convert to AndroidX Library

**Comment by shivaniaakanksha** [comment]
> @shivaniaakanksha We have to convert to AndroidX Library

@SAITARUN55  any reference on how to do this?

**Comment by SAITARUN55** [comment]
Here it is 
https://goo.gl/CP92wY
or
https://www.youtube.com/watch?v=5jbYc57lYL0
https://www.youtube.com/watch?v=ZRl4t2KdEHQ
I think flutter docs would be good enough.

**Comment by prasadsunny1** [comment]
I guess I have an idea to upgrade to AndroidX
Please come to gitter chat.

**Comment by SAITARUN55** [comment]
@prasadsunny1 Bhai , I already told @iamriya to do when she asked me 2 days back

**Comment by sakshivij** [comment]
Is this issue done ? @SAITARUN55

**Comment by SAITARUN55** [comment]
@shivaniaakanksha is doing... it takes time as this issue and #3 has to be done simultaneously

**Comment by shivaniaakanksha** [comment]
> Here it is
> https://goo.gl/CP92wY
> or
> https://www.youtube.com/watch?v=5jbYc57lYL0
> https://www.youtube.com/watch?v=ZRl4t2KdEHQ
> I think flutter docs would be good enough.

![Capture](https://user-images.githubusercontent.com/47736716/77307707-c4b79200-6d1f-11ea-9c99-c853b4971c6b.PNG)
 I tried all the methods given above but still, it's not working. @SAITARUN55 what to do next?

**Comment by SAITARUN55** [comment]
Check slack I had messaged u regarding this few hours ago as there were errors when you raised a pr on #18 So, I had reverted it

**Comment by SAITARUN55** [comment]
That's why it is showing you

**Comment by shivaniaakanksha** [comment]
> @SAITARUN55 After adding Firebase dependencies and setup for android I'am facing this Error , written below.
> Launching lib\main.dart on vivo 1601 in debug mode...
> [!] Your app isn't using AndroidX.
> To avoid potential build failures, you can quickly migrate your app by following the steps on https://goo.gl/CP92wY.
> 
> FAILURE: Build failed with an exception.
> 
> * Where:
>   Build file 'E:\flutter\development\metroapp\android\app\build.gradle' line: 24
> * What went wrong:
>   A problem occurred evaluating project ':app'.
> 
> > ASCII
> 
> * Try:
>   Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 2s
> Gradle task assembleDebug failed with exit code 1
> Exited (sigterm)

Yes, I saw the PR, but to solve the above I need to shift the project to AndroidX, I tried these methods given below  
https://goo.gl/CP92wY
or
https://www.youtube.com/watch?v=5jbYc57lYL0
https://www.youtube.com/watch?v=ZRl4t2KdEHQ

but while performing the above process this window popUps in the end.
![Capture](https://user-images.githubusercontent.com/47736716/77326131-5b477b80-6d3f-11ea-9dba-d9906cea6812.PNG)

Please Look forward to this. @SAITARUN55 @prasadsunny1

**Comment by shivaniaakanksha** [comment]
> @SAITARUN55 After adding Firebase dependencies and setup for android I'am facing this Error , written below.
> Launching lib\main.dart on vivo 1601 in debug mode...
> [!] Your app isn't using AndroidX.
> To avoid potential build failures, you can quickly migrate your app by following the steps on https://goo.gl/CP92wY.
> 
> FAILURE: Build failed with an exception.
> 
> * Where:
>   Build file 'E:\flutter\development\metroapp\android\app\build.gradle' line: 24
> * What went wrong:
>   A problem occurred evaluating project ':app'.
> 
> > ASCII
> 
> * Try:
>   Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 2s
> Gradle task assembleDebug failed with exit code 1
> Exited (sigterm)

Yes, I saw the PR, but to solve the above I need to shift the project to AndroidX, I tried these methods given below  
https://goo.gl/CP92wY
or
https://www.youtube.com/watch?v=5jbYc57lYL0
https://www.youtube.com/watch?v=ZRl4t2KdEHQ

but while performing the above process this window popUps in the end.
![Capture](https://user-images.githubusercontent.com/47736716/77326131-5b477b80-6d3f-11ea-9dba-d9906cea6812.PNG)

Please Look forward to this. @SAITARUN55 @prasadsunny1

**Comment by prasadsunny1** [comment]
status update, please.

**Comment by shivaniaakanksha** [comment]
> status update, please.

The AndroidX library is shifted now.
